### PR TITLE
feat: bunch of improvements on blackbox_exporer

### PR DESCRIPTION
**Pull request by anarcat**
This puts the config file in `/etc/prometheus`, allows changes to the binary name (so we can use the Debian package directly).

Closes: #178

### CODE REVIEW

File: roles/blackbox_exporter/defaults/main.yml [hunk 1]
```diff
@@ -71,7 +71,10 @@ blackbox_exporter_configuration_modules:
 #      validate_answer_rrs:
 #        fail_if_matches_regexp: [test]
 
-# Where to put the blackbox_exporter.yml main configuration file
-blackbox_exporter_config_dir: /etc
+# Where to put the blackbox.yml main configuration file
+blackbox_exporter_config_dir: /etc/prometheus
```
Comment (SuperQ) [inline roles/blackbox_exporter/defaults/main.yml:75]: This should be `/etc/blackbox_exporter` to match the other roles.

File: roles/blackbox_exporter/defaults/main.yml [hunk 2]
```diff
@@ -71,7 +71,10 @@ blackbox_exporter_configuration_modules:
...
 
 blackbox_exporter_binary_install_dir: "/usr/local/bin"
+blackbox_exporter_binary_name: "blackbox_exporter"
```
Comment (SuperQ) [inline roles/blackbox_exporter/defaults/main.yml:78]: I don't think we want to do this.

**Review by SuperQ — CHANGES_REQUESTED** [review]

File: roles/blackbox_exporter/defaults/main.yml (hunk 2 above)
Comment (anarcat) [reply roles/blackbox_exporter/defaults/main.yml:78]: could you expand on this? what exactly do we want to not do here, and why?

File: roles/blackbox_exporter/defaults/main.yml (hunk 1 above)
Comment (anarcat) [reply roles/blackbox_exporter/defaults/main.yml]: okay, done.

File: roles/blackbox_exporter/defaults/main.yml (hunk 2 above)
Comment (gardar) [reply roles/blackbox_exporter/defaults/main.yml:78]: I don't see any need/benefit with having the binary name customizable, we should keep the options at minimal and avoid having options just for options sake.

Comment (anarcat) [reply roles/blackbox_exporter/defaults/main.yml:78]: this is not options option's sake: i have modified this option for it to be compatible with the debian package....

Comment (gardar) [reply roles/blackbox_exporter/defaults/main.yml:78]: Ah apologies. At the moment we do not support other installation methods than the binary install method that's currently in the roles.
When and if we expand that into rpm/deb/etc. packages we would definitely want the roles to install those packages, then variables such as these would most likely be populated internally in the role and not user configurable.
With that being said, such a feature would be a major change and I would first like to take care of some streamlining/deduplicating of tasks in the roles we currently have. (#64 )

Comment (anarcat) [reply roles/blackbox_exporter/defaults/main.yml:78]: ah. well, i guess one thing i'm saying is you're almost there and you just need to use the right binary name here. i've deployed prom on a debian 12 host using this playbook and it kind of worked after the bunch of patches i had to roll here...

**Comment by gardar** [comment]
Obsoleted by #425

**Pull request closed without merging**